core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,267 @@
 package webapi
 
 import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
-	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/workflow"
 )
 
-func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (job.ServiceCtx, error) {
+const defaultSendChannelBufferSize = 1000
+
+const triggerType = "web-trigger@1.0.0"
+
+type TriggerConfig struct {
+	AllowedSenders []ethCommon.Address      `toml:"allowedSenders"`
+	Allowedtopics  []string                 `toml:"allowedTopics"`
+	RateLimiter    common.RateLimiterConfig `toml:"rateLimiter"`
+	RequiredParams []string                 `toml:"requiredParams"`
+}
+
+type Response struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"error_message,omitempty"`
+	Status       string `json:"ACCEPTED"`
+}
+
+// Handles connections to the webapi trigger
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	config    TriggerConfig
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+	mu        sync.Mutex
+	// Will this have to get pulled into a store to have the topic and workflow ID?
+	registeredWorkflows map[string]chan capabilities.TriggerResponse
+	allowedSendersMap   map[string]bool
+	signerKey           *ecdsa.PrivateKey
+	rateLimiter         *common.RateLimiter
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+// TODO: From Design doc,
+// Once connected to a Gateway, each connector handler periodically sends metadata messages containing aggregated
+// config for all registered workflow specs using web-trigger.
+
+func NewTrigger(config TriggerConfig, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*triggerConnectorHandler, error) {
 	// TODO (CAPPL-22, CAPPL-24):
 	//   - decode config
 	//   - create an implementation of the capability API and add it to the Registry
 	//   - create a handler and register it with Gateway Connector
 	//   - manage trigger subscriptions
 	//   - process incoming trigger events and related metadata
-	return nil, nil
+
+	rateLimiter, err := common.NewRateLimiter(config.RateLimiter)
```

Comment:
similar to below. `rateLimit` is also a workflow config https://docs.google.com/document/d/1mCTAo-ix-P923eUlh4SloZfBN9PCvgf90oHWbmykjsc/edit#heading=h.qqkbzbl5lbmv